### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by RicardoEgas**
**In this PR, it will be implemented the following;**

- Remove current_user method in ApplicationController and replace it with the devise's one;
- Install and setup devise that does the following:

   1. Can register a new user.
   2. User logs in with a combination of email and password.
   3. Hashed password should be stored in the database.
   4. Ask for confirmation of email.
   5. Can reset password.

- Modify the views in devise for registration and login to match the wireframes and styling.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+    <%= f.submit "Log in", class:"login-button" %>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (DammyShittu) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly design the login form to match the wireframe. The elements should be centered per the wireframe layout

  Your Implementation | Sneek Peek Design
  :---:|:---:
  log-ric | wireframe-ric

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  <div>
+    <%= f.submit "Sign up" %>
+  </div>
+
+    <%= render "devise/shared/links" %>
+<% end %>
+  </div>
+
+</div>
```
Comment (DammyShittu) [inline app/views/devise/registrations/new.html.erb]: - [ ] Kindly redesign the sign up form also.

  reg-ric

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @RicardoEgas and @M-AminAlizadeh,

## Highlights
✔️ Devise installation

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile
Comment (AbelG101) [inline Gemfile:1]: - [ ] [OPTIONAL] The previous reviewer forgot to mention this but please make sure you have the proper linters config setup in your project. I can see you only have 1 check running it should instead be 3 checks (with the addition of Stylelint and node_modules checker) as shown below 
  image
- You can find the Rails [linter config here](https://github.com/microverseinc/linters-config/tree/master/ror) 😉

**Review by AbelG101 — APPROVED** [review]
# Hi @RicardoEgas 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Login layout matches wireframe
- Correct use of Git Flow ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- Check the comment below...

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**